### DISCUSSION TITLE
Error in installing LAMMPS with Allegro

### DISCUSSION THREAD

**Discussion by kuntalg97** [mir-group/allegro#98] [Q&A]
Hi,

I'm having some issues in installing LAMMPS with pair_style allegro. After using the make command, while the .cpp files are being compiled, I get this error:

.
.
.
[ 81%] Building CXX object CMakeFiles/lammps.dir/project2/kuntalg/Softwares/Allegro/lammps/src/pair.cpp.o
[ 81%] Building CXX object CMakeFiles/lammps.dir/project2/kuntalg/Softwares/Allegro/lammps/src/pair_allegro.cpp.o
/project2/kuntalg/Softwares/Allegro/lammps/src/pair_allegro.cpp: In member function ‘virtual void LAMMPS_NS::PairAllegro<precision>::compute(int, int)’:
/project2/kuntalg/Softwares/Allegro/lammps/src/pair_allegro.cpp:513:21: error: ‘class std::map<std::basic_string<char>, at::Tensor>’ has no member named ‘insert_or_assign’
  513 |       custom_output.insert_or_assign(output_name, output.at(output_name).toTensor().detach());
      |                     ^~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/lammps.dir/build.make:4150: CMakeFiles/lammps.dir/project2/kuntalg/Softwares/Allegro/lammps/src/pair_allegro.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:171: CMakeFiles/lammps.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

I'm using cuda/11.3, and other modules such as mkl, intel, intelmpi, gcc, gsl are already loaded. I'm using PyTorch 1.11.0 and it can recognize cuda on my HPC (I'm installiing on an interactive GPU node).

Any suggestions on how to fix this and successfully compile LAMMPS with Allegro?

Thank you so much!
Kuntal

**Comment by Linux-cpp-lisp** [comment]
Please open a new issue on https://github.com/mir-group/pair_nequip_allegro if this issue remains after the recent major updates.
